src/Nethermind/Nethermind.Mev/Source/BundlePool.cs

```diff
@@ -169,52 +215,40 @@ private void OnNewSuggestedBlock(object? sender, BlockEventArgs e)
             }
         }
 
-        private void OnBlocksFinalized(object? sender, FinalizeEventArgs e)
+        private void ResortBundlesByBlock(long newBlockNumber)
         {
-            long maxFinalizedBlockNumber = e.FinalizedBlocks.Select(b => b.Number).Max();
-            if (_bundles.Count > 0)
+            IEnumerable<long> Range(long start, long count)
             {
-                lock (_bundles)
+                for (long i = start; i < start + count; i++)
                 {
-                    if (_bundles.Count > 0)
-                    {
-                        MevBundle bundle = _bundles.Keys[0];
-                        while (bundle.BlockNumber <= maxFinalizedBlockNumber)
-                        {
-                            ConcurrentBag<Keccak> blocksBag = _bundles.Values[0];
-                            foreach (Keccak blockHash in blocksBag)
-                            {
-                                if (_simulatedBundles.TryGetValue(blockHash, out ConcurrentDictionary<MevBundle, SimulatedMevBundleContext>? bundleDictionary))
-                                {
-                                    if (bundleDictionary.TryRemove(bundle, out SimulatedMevBundleContext? context))
-                                    {
-                                        context.CancellationTokenSource.Cancel();
-                                        context.Dispose();
-                                    }
-
-                                    if (bundleDictionary.Count == 0)
-                                    {
-                                        _simulatedBundles.TryRemove(blockHash, out _);
-                                    }
-                                }
-                            }
+                    yield return i;
+                }
+            }
+            
+            long previousBestSuggested = _compareMevBundleByBlock!.BestBlockNumber;
+            long fromBlockNumber = Math.Min(newBlockNumber, previousBestSuggested);
+            long blockDelta = Math.Abs(newBlockNumber - previousBestSuggested);
+            _bundles.NotifyChange(Range(fromBlockNumber, blockDelta), () => _compareMevBundleByBlock.BestBlockNumber = newBlockNumber);
+        }
 
-                            _bundles.RemoveAt(0);
-                            
-                            if (_bundles.Count > 0)
-                            {
-                                bundle = _bundles.Keys[0];
-                            }
-                            else
-                            {
-                                break;
-                            }
-                        }
-                    }
+        private void OnBlocksFinalized(object? sender, FinalizeEventArgs e) //NEED TO ADD ANYTHING ELSE?
+        {
```

Comment:
I think we do not need it

Reply:
So initial idea was to use it for better (potentially faster) bundle eviction. Its debatable if we should keep it.

Reply:
We dropped it